### PR TITLE
Onboard openshift/boilerplate

### PR DESCRIPTION
**Pull request by arjunrn**
Fixed linting issues as well.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,7 +9,7 @@ require (
 	github.com/openshift/api v3.9.1-0.20190924102528-32369d4db2ad+incompatible
 	github.com/openshift/cluster-network-operator v0.0.0-20190207145423-c226dcab667e // indirect
 	github.com/openshift/hive v0.0.0-20200210203046-8b1c393442db
-	github.com/openshift/operator-custom-metrics v0.3.1-0.20200901174648-463079905232
+	github.com/openshift/operator-custom-metrics v0.4.0
```
Comment (2uasimojo) [inline go.mod:12]: This seems like a separate, unrelated change.

File: Makefile [hunk 2]
```diff
@@ -1,34 +1,6 @@
-SHELL := /usr/bin/env bash
-OPERATOR_DOCKERFILE = build/Dockerfile
+include boilerplate/generated-includes.mk
 
-# Include shared Makefiles
-include project.mk
-include standard.mk
```
Comment (2uasimojo) [inline Makefile:6]: This commit should delete these files.

**Review by 2uasimojo — COMMENTED** [review]
This is looking good. A couple of minor things noted inline.

File: go.mod (hunk 1 above)
Comment (arjunrn) [reply go.mod:12]: But without it the build failed because that commit is now it's own tag. Do you want it in a separate PR?

File: Makefile (hunk 2 above)
Comment (arjunrn) [reply Makefile:6]: Do you mean they will not be included in the `generated-includes.mk` file anymore.

Comment (2uasimojo) [reply Makefile:6]: `generated-includes.mk` brings in the `project.mk` and `standard.mk` from the _convention_. These are obsolete. That's one of the main things we've actually got working right now :)

File: go.mod (hunk 1 above)
Comment (2uasimojo) [reply go.mod:12]: Sigh, I see.

Yeah, it really should go in a separate PR, which I guess needs to be in front of this one.

File: Makefile (hunk 2 above)
Comment (arjunrn) [reply Makefile:6]: I don't understand. Do you mean that the files in the `boilerplate` directory should be deleted? Because I still see them in the boilerplate repo.

Comment (2uasimojo) [reply Makefile:6]: No. The ones in the repo root. The last two here:

```
[efried@efried deadmanssnitch-operator]$ git ls-files | grep mk$
boilerplate/_lib/boilerplate.mk
boilerplate/generated-includes.mk
boilerplate/openshift/golang-osd-operator/project.mk
boilerplate/openshift/golang-osd-operator/standard.mk
project.mk
standard.mk
```

File: go.mod (hunk 1 above)
Comment (arjunrn) [reply go.mod]: It builds without this change as well. I'll create a separate PR anyway.

File: Makefile (hunk 2 above)
Comment (arjunrn) [reply Makefile:6]: Got it. Thanks. Removed.

**Review by mwoodson — APPROVED** [review]
